tutorials/how-to-implement-rag-generativeapis/index.mdx

```diff
@@ -70,288 +76,275 @@ Create a .env file and add the following variables. These will store your API ke
+- You have the right [IAM permissions](/identity-and-access-management/iam/reference-content/policy/), specifically **ContainersRegistryFullAccess**, **ContainersFullAccess** 
+- You created an [IAM API Key capable of interacting with Object Storage](https://www.scaleway.com/en/docs/identity-and-access-management/iam/api-cli/using-api-key-object-storage/).
+- You stored the right credentials in your `.env` file, especially credentials allowing to connect to your [Managed Database Instance with admin rights](https://www.scaleway.com/en/docs/managed-databases/postgresql-and-mysql/how-to/add-users/).
 
-By integrating Scaleway Object Storage, Managed Database for PostgreSQL with pgvector, and LangChain’s embedding tools, you have the foundation to build a powerful RAG system that scales with your data while offering robust information retrieval capabilities. This approach equips you with the tools necessary to handle complex queries and deliver accurate, relevant results efficiently.
+If you encounter the following error message, try corresponding solutions:
 
-With ongoing refinement and adaptation, your RAG system can evolve to meet the changing needs of your users, ensuring that it remains a valuable asset in your AI toolkit.
\ No newline at end of file
+- `botocore.exceptions.ClientError: An error occurred (SignatureDoesNotMatch) when calling the ListObjectsV2 operation: The request signature we calculated does not match the signature you provided. Check your key and signing method.`
+  - **Solution:** Ensure your `SCW_BUCKET_NAME`, `SCW_REGION`, `SCW_BUCKET_ENDPOINT` and `SCW_SECRET_KEY` are properly configured, that corresponding IAM Principal have the proper rights and that your [IAM API Key can interact with Object Storage]
+- `urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)>`
+  - On MacOS, ensure your Python version has loaded certificates trusted by MacOS. You can fix this by `Applications/Python 3.X` (where X is your version number), and double click on `Certificates.command`.
+- `ERROR:root:An error occurred: bge-multilingual-gemma2 is not a local folder and is not a valid model identifier listed on 'https://huggingface.co/models'
+If this is a private repository, make sure to pass a token having permission to this repo either by logging in with `huggingface-cli login` or by passing `token=<your_token>``
+  - This is caused by Langchain OpenAI adapter trying to tokenize content. Ensure you set `check_embedding_ctx_length=False` in OpenAIEmbedding configuration to avoid tokenizing content (as tokenization will be performed server-side, in Generative APIs).
```

Comment:
```suggestion
**Error**: `ERROR:root:An error occurred: bge-multilingual-gemma2 is not a local folder and is not a valid model identifier listed on 'https://huggingface.co/models'
If this is a private repository, make sure to pass a token having permission to this repo either by logging in with `huggingface-cli login` or by passing `token=<your_token>``

**Solution**: This is caused by the LangChain OpenAI adapter trying to tokenize content. Ensure you set `check_embedding_ctx_length=False` in OpenAIEmbedding configuration to avoid tokenizing content, as tokenization will be performed server-side in Generative APIs.
```